### DISCUSSION TITLE
Postbox collection times in Switzerland

### DISCUSSION THREAD

**Discussion by chjohn22** [streetcomplete/StreetComplete#2963] [Feedback]
The Swiss Post did change the collection times of most postboxes in Switzerland in the beginning of June 2021. Is it possible to force a resurvey of all postboxes in Switzerland? Many boxes are not collected on Sunday or Saturday anymore and this would definitely help to change the data in an efficient way. @westnordost 

https://www.srf.ch/news/panorama/neue-leerungszeiten-die-meisten-briefkaesten-werden-am-sonntag-nicht-mehr-geleert

**Comment by matkoniecz** [comment]
Unable to read this article, but in such case I would consider deleting that info for all post boxes in Switzerland

Obviously, it would need to be discussed with mappers there and get a very clear agreement 

See https://wiki.openstreetmap.org/wiki/Automated_Edits_code_of_conduct and https://wiki.openstreetmap.org/wiki/Contact_channels

**Comment by kmpoppe** [comment]
Solving this program-wise (voiding every check_date before 2021-06-01 in Switzerland only) would reduce the number of edits, wouldn't it (and just get back to "are these collection times correct" rather than getting times that might NOT have changed)?

**Reply by chjohn22** [reply]
It seems like the collection times of allmost all postboxes changed. There are approximately 14414 postboxes in Switzerland and a great majority of them has now different collection times.

`Demnach leert die Post schweizweit am Samstag noch fast 11'400 Briefkästen, am Sonntag 1724. Künftig werden es am Samstag noch 4009 sein, am Sonntag 290. Doch auch werktags kommt es laut der Zeitschrift zum Serviceabbau. Gemäss Wertheimer wird bei 2900 Briefkästen die Leerung vom Abend auf die Zustelltour verlegt.`

https://www.luzernerzeitung.ch/news-service/wirtschaft/serviceabbau-post-leert-ihre-gelben-briefkaesten-kuenftig-weniger-haeufig-ld.2141803

**Reply by matkoniecz** [reply]
It would also reduce how long OSM has incorrect data (removed now rather on nearest resurvey which may not happen for years)

> would reduce the number of edits,

this is not really important (opinions may vary, some people are against bot edits, especially minor, solely because it requires editing)

**Reply by kmpoppe** [reply]
I rest my case your Honours ;-)

**Reply by mnalis** [reply]
I also think it would be _**preferable to remove such incorrect/known obsolete data by automated edit**_ (after getting a buy-in from the community, of course!) 
Not only would it remove incorrect data immediately (instead of it remaining in database for years or decades), but it would not require adding (and keeping for indefinite time!) tons of various "exceptions" in the SC code (as those would keep adding up with time) - which in addition to being dead cruft which would ideally need to be checked every now and then if it is still needed, would be confusing to the average user (ie. _"it asks me for the quest which was already answered few months ago!"_) .

**Reply by timothywashere** [reply]
Doing a mass delete would trigger the resurvey but also delete any that are correct since the changes affect most, but not all post boxes. And some have already been surveyed correctly since the change.

**Reply by mnalis** [reply]
@timothywashere Still deleting looks preferable to me - in that case, all data that remain is either correct or missing, and missing data is trivially identifiable and thus added by interesting mappers (either via SC, or other means). And the change needs to be done only once - in highly specific automated edit. 

If instead all data is left as-is, majority of it will be incorrect,  and with time only parts of it will be fixed. Missing information is (IMHO) much less of a problem then incorrect information, especially if it is not trivially and automatically known which information is correct, and which is not. And the complex logic (see below) to determine what needs to be updated would need to be added to all programs that are used for mapping, not just StreetComplete (but also Vespucci, OsmAnd etc.). Additionally, as such change is unlikely to happen in _all_ editing tools, the rate of fixing incorrect information would likely be much slower that the rate of filling in missing information (where all tools could participate as usual). 

As for your other comment elsewhere in the thread: note that if some data has already been correctly surveyed after the change, automated edit needs not remove it (automated edit can be programed for other exceptions too - for example if postboxes inside shopping malls have not changed the collecting times, it can be checked for, and collection times of those postboxes left alone, etc.)

**Comment by eginhard** [comment]
I had a brief look at how to implement this properly, e.g. with a date entry in `res/country_metadata/` when the collection times in Switzerland should be resurveyed. But that seems like a very specific solution for a rare scenario and it's probably not worth merging.

Also, there is currently no separate resurvey form for collection times (#2105), so in StreetComplete there would be no difference anyway to deleting all `collection_times` values before that date.

**Comment by timothywashere** [comment]
Royal Mail in the UK have just changed the last collection times of pretty much all their post boxes. Basically they are going to be collected by the postman on his rounds instead of a man in a van.

If such a reset were possible, it would be nice to use in the UK too.  May be in other countries in the future too given the drastic changes in the postal industry the last few years are probably world wide.

**Reply by westnordost** [reply]
I've read the switzerland thread above,  seems the conclusion was that it is better too remove the wrong information at the time the change is made.

**Reply by timothywashere** [reply]
May I ask what the SC resurvey interval is for postbox colleciton times?

I don't think doing a mass delete in the UK is viable but I will consult on the UK mailing list. It seems to affect different areas of UK at different dates and some boxes are not affected at all (the big ones in shopping areas etc)

**Reply by westnordost** [reply]
2 years
